src/main/java/uk/gov/hmcts/ccd/domain/types/CaseDataValidator.java

```diff
@@ -26,14 +26,17 @@
     private static final String FIELD_SEPARATOR = ".";
 
     private List<FieldValidator> validators;
+    private ValidationContext validationContext;
 
     @Inject
     public CaseDataValidator(final List<FieldValidator> validators) {
         this.validators = validators;
     }
 
     public List<ValidationResult> validate(final Map<String, JsonNode> data,
-                                           final List<CaseFieldDefinition> caseFieldDefinitions) {
+                                           final List<CaseFieldDefinition> caseFieldDefinitions,
+                                           final ValidationContext validationContext) {
+        this.validationContext = validationContext;
```

Comment:
this class is a Singleton, so it's shared between many threads. It's not safe to set the validation context in an instance variable.
You need to pass that to the validate method:
`validate(data, caseFieldDefinitions, CaseDataValidator.EMPTY_STRING, validationContext);

Reply:
Yes , DONE